pkg/cluster/internal/delete/delete.go

```diff
@@ -33,17 +33,19 @@ func Cluster(logger log.Logger, p providers.Provider, name, explicitKubeconfigPa
 		return errors.Wrap(err, "error listing nodes")
 	}
 
-	kerr := kubeconfig.Remove(name, explicitKubeconfigPath)
+	kerr := kubeconfig.Remove(logger, name, explicitKubeconfigPath)
 	if kerr != nil {
 		logger.Errorf("failed to update kubeconfig: %v", kerr)
+		return kerr
```

Comment:
@BenTheElder we don't try to delete the nodes anymore?

Reply:
That's a good point. While it's not great if kubeconfig can't be updated, we probably shouldn't just error out completely at that point.

I could see either holding on to `kerr` and adding one more check after the `DeleteNodes` block is complete:

```go
if kerr != nil {
    return kerr
}

return nil
```

Or just ignoring it and going with just the logging of the error.

The current code uses the former approach. We should probably stick with that.

Reply:
Is there anything I need to change or is this good enough as is?

If you look through the history of this PR, I had made some changes to this but I ended up putting it back.

Reply:
I think I would prefer the change I noted above. I think that's what @aojea was alluding to as well.

Reply:
What about the `return nil` down on line 50?

Reply:
Maybe I don't understand. That's already in place.

Reply:
- [ ] Delete line 39
- [ ] Insert the following before the `return nil` at the end:

```go
if kerr != nil {
    return kerr
}
```

Reply:
Alright done. But what about the `return err` on line 44?

Reply:
True, otherwise if their is a failure removing the kubeconfig, then the node deletion fails, they never see that. Though then they would still need to fix the issue and clean things up anyway.

Maybe it would be better to move the kubeconfig deletion completely after the node deletion? That way if the kubeconfig deletion fails, at least the cluster itself has already been removed and the user would just need to clean up their kubeconfig.

I think either way is fine. Let's see if others have a stronger opinion on this.

Otherwise looks great - thanks for the updates!

Reply:
@aojea @BenTheElder any updates here?